pkg/hooks/loader.go

```diff
@@ -342,47 +334,21 @@ func (h *Hook) findAndCollectChildProcesses(parentPID string, pids *[]int) {
 	}
 }
 
-// Stop and Remove the docker container
-func (h *Hook) StopAndRemoveDockerContainer() {
-	dockerClient := h.idc
-	containerID := h.dockerDetails.containerID
-
-	container, err := dockerClient.ContainerInspect(context.Background(), containerID)
-    if err != nil {
-        fmt.Println(err)
-        return
-    }
-
-    if container.State.Status == "running" {
-        err = dockerClient.ContainerStop(context.Background(), containerID, dockerContainerPkg.StopOptions{})
-        if err != nil {
-            fmt.Println(err)
-            return
-        }
-    }
-
-	removeOptions := types.ContainerRemoveOptions{
-		RemoveVolumes: true,
-		Force:         true,
-	}
-
-    err = dockerClient.ContainerRemove(context.Background(), containerID, removeOptions)
-    if err != nil {
-        fmt.Println(err)
-        return
-    }
-}
-
 // StopUserApplication stops the user application
 func (h *Hook) StopUserApplication() {
 	if h.userAppCmd != nil && h.userAppCmd.Process != nil {
 		h.logger.Debug("the process state for the user process", zap.String("state", h.userAppCmd.ProcessState.String()), zap.Any("processState", h.userAppCmd.ProcessState))
 		if h.userAppCmd.ProcessState != nil && h.userAppCmd.ProcessState.Exited() {
 			return
 		}
-		
-		// Stop Docker Container and Remove it
-		h.StopAndRemoveDockerContainer()
+
+		// Stop Docker Container and Remove it if Keploy ran using docker
+		if len(h.idc.GetContainerID()) != 0  {
+			err := h.idc.StopAndRemoveDockerContainer()
+			if err != nil {
```

Comment:
This should be an error log saying failed to stop container/ remove container , zap the error and suggest user to delete it manually